src/routes/commercialRegistrations/search.ts

```diff
@@ -0,0 +1,45 @@
+import type { Request, Response } from "express";
+import { businesses } from "../../fixtures/businesses";
+import { SearchQuery } from "./types/searchQuery";
+import { Registration } from "./types/registration";
+import { issuerNames } from "../../fixtures/issuerNames";
+
+export type SearchRequest = Partial<Request<{}, {}, {}, SearchQuery>>;
+export type SearchResponse = Partial<Response<Registration[] | []>>;
+
+/**
+ * @see {@link https://docs.solarisgroup.com/api-reference/onboarding/businesses/#tag/Business-Registrations/paths/~1v1~1commercial_registrations~1search_by_name/get}
+ *
+ * Searches for businesses by name and country
+ */
+export const search = (req: SearchRequest, res: SearchResponse) => {
+  const { country = "DE", name } = req.query;
+  const foundBusinesses = businesses.filter(
+    (business) =>
+      String(business.name)
+        .toLowerCase()
+        .includes(String(name).toLowerCase()) &&
+      business.address.country === country
+  );
+  const shouldGenerateBusiness =
+    String(name).length >= 4 && String(name).length <= 10;
```

Comment:
the name with a length over 10 characters won't be found and we can test empty suggestions list, right?

Reply:
True